### PR TITLE
Add the `resampling-filter` Imagine option

### PR DESCRIPTION
**Pull request by ausi**
~~Should be merged after https://github.com/contao/image/pull/108~~ ✅

### CODE REVIEW

File: core-bundle/tests/DependencyInjection/ConfigurationTest.php [hunk 1]
```diff
@@ -292,7 +292,7 @@ private function checkKeys(array $configuration): void
                 $this->checkKeys($prototype->getChildren());
             }
 
-            if (\is_string($key) && !$value->isDeprecated()) {
+            if (\is_string($key) && !$value->isDeprecated() && 'resampling-filter' !== $key) {
```
Comment (ausi) [inline core-bundle/tests/DependencyInjection/ConfigurationTest.php:295]: See https://github.com/contao/image/pull/108#discussion_r1784113131
I think we should use the original name as it is used by Imagine, WDYT?

Comment (leofeyer) [reply core-bundle/tests/DependencyInjection/ConfigurationTest.php:295]: I agree.

File: core-bundle/src/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -184,8 +184,12 @@ private function addImageNode(): NodeDefinition
                             ->info('Allows to disable the layer flattening of animated images. Set this option to false to support animations. It has no effect with Gd as Imagine service.')
                         ->end()
                         ->scalarNode('interlace')
+                            ->info('One of the Imagine\Image\ImageInterface::INTERLACE_* constants.')
                             ->defaultValue(ImageInterface::INTERLACE_PLANE)
                         ->end()
+                        ->scalarNode('resampling-filter')
+                            ->info('Filter used when downsampling images. One of the Imagine\Image\ImageInterface::FILTER_* constants. It has no effect with Gd or SVG as Imagine service.')
+                        ->end()
```
Comment (leofeyer) [inline core-bundle/src/DependencyInjection/Configuration.php:192]: No `->defaultValue()` here?

Comment (ausi) [reply core-bundle/src/DependencyInjection/Configuration.php:192]: The contao/image library already uses a good default, so I would not add one here. Adding a default would change the hash for all images and that is not OK for a bugfix release IMO. If we want to add a default we would have to add it in Contao 5.5, but I don’t think we need that.

File: core-bundle/src/DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -184,8 +184,12 @@ private function addImageNode(): NodeDefinition
                             ->info('Allows to disable the layer flattening of animated images. Set this option to false to support animations. It has no effect with Gd as Imagine service.')
                         ->end()
                         ->scalarNode('interlace')
+                            ->info('One of the Imagine\Image\ImageInterface::INTERLACE_* constants.')
                             ->defaultValue(ImageInterface::INTERLACE_PLANE)
                         ->end()
+                        ->scalarNode('resampling-filter')
```
Comment (leofeyer) [inline core-bundle/src/DependencyInjection/Configuration.php:190]:
```suggestion
                        ->scalarNode('resampling_filter')
```

All other keys use underscores, so maybe this one should, too?

Comment (ausi) [reply core-bundle/src/DependencyInjection/Configuration.php:190]: This comes from Imagine. They use underscores for most options, but for some they use dashes 🙈
See https://github.com/contao/contao/pull/7595#discussion_r1784132093

Comment (leofeyer) [reply core-bundle/src/DependencyInjection/Configuration.php:190]: I see. And I didn‘t realize that we pass these keys directly to the library.

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @ausi.

**Pull request merged**